great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -305,36 +324,36 @@ def _get_domains(
         )
         candidate_column_names.extend(allowed_column_names_passthrough)
 
-        column_name: str
         domains: List[Domain] = build_domains_from_column_names(
             rule_name=rule_name,
             column_names=candidate_column_names,
             domain_type=self.domain_type,
-            table_column_name_to_inferred_semantic_domain_type_map=self.semantic_type_filter.table_column_name_to_inferred_semantic_domain_type_map,
+            table_column_name_to_inferred_semantic_domain_type_map=self.semantic_type_filter.table_column_name_to_inferred_semantic_domain_type_map,  # type: ignore[operator,union-attr] # could be None
         )
 
         return domains
 
     def _generate_metric_configurations_to_check_cardinality(
         self,
-        batch_ids: List[str],
         column_names: List[str],
+        batch_ids: Optional[List[str]] = None,
```

Comment:
This seems like a somewhat dangerous move. Changing the order of the parameters and changing one from a required positional large to an optional keyword argument.

If rule_based_profiler was more heavily type-checked I would feel better about this, but right now a lot of the package is being exlcuded.

Reply:
@Kilo59 This is a private method -- and while most of the package is excluded, this module has been moved along quite a bit.  We should accept inching toward improvements, right?

Reply:
If this is only used within the module or you are reasonably sure of the safety of this move I'm fine with it.